packages/cli/commands/import/credentials.ts

```diff
@@ -56,60 +74,111 @@ export class ImportCredentialsCommand extends Command {
 		try {
 			await Db.init();
 
+			await this.initOwnerCredentialRole();
+			const user = flags.id ? await this.getAssignee(flags.id) : await this.getOwner();
+
 			// Make sure the settings exist
 			await UserSettings.prepareUserSettings();
-			let i;
+			let i: number;
 
 			const encryptionKey = await UserSettings.getEncryptionKey();
 			if (encryptionKey === undefined) {
-				throw new Error('No encryption key got found to encrypt the credentials!');
+				throw new Error('No encryption key found to encrypt the credentials!');
 			}
 
 			if (flags.separate) {
 				const files = await glob(
 					`${flags.input.endsWith(path.sep) ? flags.input : flags.input + path.sep}*.json`,
 				);
 				for (i = 0; i < files.length; i++) {
-					// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 					const credential = JSON.parse(fs.readFileSync(files[i], { encoding: 'utf8' }));
 
-					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
 					if (typeof credential.data === 'object') {
 						// plain data / decrypted input. Should be encrypted first.
-						// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
 						Credentials.prototype.setData.call(credential, credential.data, encryptionKey);
 					}
 
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Credentials!.save(credential);
+					await this.storeCredential(credential, user);
 				}
 			} else {
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-				const fileContents = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
+				const credentials = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
 
-				if (!Array.isArray(fileContents)) {
-					throw new Error(`File does not seem to contain credentials.`);
+				if (!Array.isArray(credentials)) {
+					throw new Error(
+						'File does not seem to contain credentials. Make sure the credentials are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
-					if (typeof fileContents[i].data === 'object') {
+				for (i = 0; i < credentials.length; i++) {
+					if (typeof credentials[i].data === 'object') {
 						// plain data / decrypted input. Should be encrypted first.
-						Credentials.prototype.setData.call(
-							fileContents[i],
-							fileContents[i].data,
-							encryptionKey,
-						);
+						Credentials.prototype.setData.call(credentials[i], credentials[i].data, encryptionKey);
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Credentials!.save(fileContents[i]);
+					await this.storeCredential(credentials[i], user);
```

Comment:
let's wrap this part in a transaction and either set `this.transactionManager` or pass it as a param to `this.storeCredential` 👍🏼